HeterogeneousCore/SonicTriton/interface/TritonService.h

```diff
@@ -45,17 +49,31 @@ class TritonService {
     int wait;
     std::string instanceName;
     std::string tempDir;
+    std::string imageName;
+    std::string sandboxName;
   };
   struct Server {
     Server(const edm::ParameterSet& pset)
         : url(pset.getUntrackedParameter<std::string>("address") + ":" +
               std::to_string(pset.getUntrackedParameter<unsigned>("port"))),
-          isFallback(pset.getUntrackedParameter<std::string>("name") == fallbackName) {}
-    Server(const std::string& name_, const std::string& url_) : url(url_), isFallback(name_ == fallbackName) {}
+          isFallback(pset.getUntrackedParameter<std::string>("name") == fallbackName),
+          type(TritonServerType::Remote),
+          useSsl(pset.getUntrackedParameter<bool>("useSsl")) {
+      if (useSsl) {
+        sslOptions.root_certificates = pset.getUntrackedParameter<std::string>("rootCertificates");
+        sslOptions.private_key = pset.getUntrackedParameter<std::string>("privateKey");
+        sslOptions.certificate_chain = pset.getUntrackedParameter<std::string>("certificateChain");
+      }
+    }
+    Server(const std::string& name_, const std::string& url_, TritonServerType type_)
+        : url(url_), isFallback(name_ == fallbackName), type(type_), useSsl(false) {}
 
     //members
     std::string url;
     bool isFallback;
+    TritonServerType type;
+    bool useSsl;
```

Comment:
```suggestion
    bool useSsl;
    TritonServerType type;
```
to save (likely) 8 bytes